provisioning/device/src/ProvisioningRegistrationAdditionalData.cs

```diff
@@ -11,8 +10,15 @@ namespace Microsoft.Azure.Devices.Provisioning.Client
     public class ProvisioningRegistrationAdditionalData
     {
         /// <summary>
-        /// Additional (optional) Json Data to be sent to the service 
+        /// Additional (optional) Json Data to be sent to the service
         /// </summary>
         public string JsonData { get; set; }
+
+        /// <summary>
+        /// The PEM encoded operational certificate request that DPS will send to its linked CA and the CA will sign and return an X509 device identity operational client certificate to the device.
+        /// DPS will register the device and operational client certificate thumbprint in IoT Hub and return the certificate to the IoT device.
+        /// The IoT device can then use the operational certificate to authenticate with IoT Hub.
+        /// </summary>
+        public string OperationalCertificateRequest { get; set; }
```

Comment:
This feels like an odd spot to put this new field. This class, as I understand it, is for user defined data only and for users who are using custom webhooks for provisioning. Maybe this class could be renamed in v2 ("RegistrationOptions" or something), but right now this feels misplaced

Reply:
> This class, as I understand it, is for user defined data only and for users who are using custom webhooks for provisioning.

Is that so? From the name I inferred that this class was a holder for any additional "data" that accompanies the registration request. That could be a payload containing custom webhook data or the pnp model data, and/or in this case, the certificate signing request.

I agree that calling this `RegistrationOptions` would have been a better option that we can definitely include in v2.

Reply:
Technically, even the certificate request is a part of the "json data", since it is all json at the end. So, I can see how the current structure can get confusing. 
But other than adding another overload and a new type, this seemed like the better of the two options.
What do you think?

Reply:
Your current solution is fine for v1, yeah. I'd definitely like to see this class renamed in v2 though if we go this direction

Reply:
> Is that so? From the name I inferred that this class was a holder for any additional "data" that accompanies the registration request. That could be a payload containing custom webhook data or the pnp model data, and/or in this case, the certificate signing request.

The user defined json object here does nothing unless a user is using a custom allocation policy to decide which hub a given device is assigned to. By contrast, your new field will be used regardless of what allocation policy is attached to an enrollment. At the very least, I think it's worthwhile to call this out in the header docs